solutions/client-landing-zone/client-folder/standard/applications-infrastructure/host-project/network/public-dns.yaml

```diff
@@ -0,0 +1,101 @@
+  resourceID: client-name-standard-public-dns # kpt-set: ${client-name}-standard-public-dns
+  dnsName: "dns-name" # kpt-set: ${dns-name}
+  visibility: public
+  dnssecConfig:
+    state: "on"
+  cloudLoggingConfig:
+    enableLogging: true
+---
+# NS Record Set for client DNS subzone
+# This will update the client DNS subzone NS record with a specific set of nameservers, configured in setters.
+# SC-22
+apiVersion: dns.cnrm.cloud.google.com/v1beta1
+kind: DNSRecordSet
+metadata:
+  name: client-name-standard-public-dns-ns-rset # kpt-set: ${client-name}-standard-public-dns-ns-rset
+  namespace: client-name-networking # kpt-set: ${client-name}-networking
+  annotations:
+    cnrm.cloud.google.com/project-id: host-project-id # kpt-set: ${host-project-id}
+    config.kubernetes.io/depends-on: dns.cnrm.cloud.google.com/namespaces/client-name-networking/DNSManagedZone/client-name-standard-public-dns # kpt-set: dns.cnrm.cloud.google.com/namespaces/${client-name}-networking/DNSManagedZone/${client-name}-standard-public-dns
+spec:
+  name: "dns-name" # kpt-set: ${dns-name}
+  type: "NS"
+  ttl: 21600
+  managedZoneRef:
+    name: client-name-standard-public-dns # kpt-set: ${client-name}-standard-public-dns
+    namespace: client-name-networking # kpt-set: ${client-name}-networking
+  rrdatas: # kpt-set: ${dns-nameservers}
+    - "ns-cloud-a1.googledomains.com."
+    - "ns-cloud-a2.googledomains.com."
+    - "ns-cloud-a3.googledomains.com."
+    - "ns-cloud-a4.googledomains.com."
+---
+# SOA Record Set for Client DNS subzone
+# This will update the client DNS subzone SOA's primary name server, configured in setters.
+# SC-22
+apiVersion: dns.cnrm.cloud.google.com/v1beta1
+kind: DNSRecordSet
+metadata:
+  name: client-name-standard-public-dns-soa-rset # kpt-set: ${client-name}-standard-public-dns-soa-rset
+  namespace: client-name-networking # kpt-set: ${client-name}-networking
+  annotations:
+    cnrm.cloud.google.com/project-id: host-project-id # kpt-set: ${host-project-id}
+    config.kubernetes.io/depends-on: dns.cnrm.cloud.google.com/namespaces/client-name-networking/DNSManagedZone/client-name-standard-public-dns # kpt-set: dns.cnrm.cloud.google.com/namespaces/${client-name}-networking/DNSManagedZone/${client-name}-standard-public-dns
+spec:
+  name: "dns-name" # kpt-set: ${dns-name}
+  type: "SOA"
+  ttl: 21600
+  managedZoneRef:
+    name: client-name-standard-public-dns # kpt-set: ${client-name}-standard-public-dns
+    namespace: client-name-networking # kpt-set: ${client-name}-networking
+  rrdatas: # kpt-set: ${dns-primary-nameserver}
+    - "ns-cloud-a1.googledomains.com. cloud-dns-hostmaster.google.com. 1 21600 3600 259200 300"
+---
+# Record Set for client DNS subzone in public core DNS zone
+# SC-22
+apiVersion: dns.cnrm.cloud.google.com/v1beta1
+kind: DNSRecordSet
+metadata:
+  name: client-name-standard-core-public-dns-ns-rset # kpt-set: ${client-name}-standard-core-public-dns-ns-rset
+  namespace: networking # kpt-set: networking
```

Comment:
we have an new internal bug 5804 to try to deploy all client resources with client namespaces, should we add this in?

Reply:
This resource is being deployed in tier 1 and is part of the core-landing-zone package so I believe it should reside within the core namespaces (hierarchy, projects, networking, etc).

Reply:
Actually, that's incorrect. The record set is part of the managed dns zone from tier 1 so I thought of using the same namespace to keep everything from the core dns zone together.

Reply:
ok, either way I added it to the bug for tracking
FYI for @davelanglois-ssc

Reply:
Working on it now.

Reply:
Fixed/Changed

Reply:
@lucstjean-ssc  just missing this : the client-setup package needs to be customized to grant the role Tier2 DNS Record Admin

Reply:
@davelanglois-ssc  There's a DNS Admin role in the client-name-networking namespace file. Should we keep/remove or replace with this Tier 2 DNS Record Admin custom role?

Reply:
so we need both

Reply:
I opened new conversation on the specific lines where I have made comments.